tests/test_frame.py

```diff
@@ -270,6 +270,67 @@ def test_encode_error_type(self, frame, value):
         with pytest.raises(ValueError):
             frame.encode({'MotorSpeed': value})
 
+@pytest.mark.unit
+class TestEncodeDecodeArray:
+    """Test encode/decode of signal with array values"""
+    @pytest.mark.parametrize("use_converter", [True, False])
+    def test_encode_decode_array(self, use_converter):
+        signal = LinSignal('BattCurr', 24, [0, 0, 2])
+        encoding_type = LinSignalEncodingType(
+            "BattCurrCoding",
+            [
+                PhysicalValue(0, 182272, 0.00390625, -512, "A"),
```

Comment:
In a previous commit this made a lot more sense to me, there the physical value encoder simply handled the array signal values as they were completely ordinary values ranging from 0-255, this is why I suggested moving support into the Physical and Logical as that could be a bit cleaner.